docs: include a section with CircleCI instructions (#318)
game-ci/documentation: docs/11-circleci/04-test.mdx

---
toc_max_heading_level: 4
---

# Test

> The Unity orb provides the
> [test job](https://circleci.com/developer/orbs/orb/game-ci/unity#jobs-test) to facilitate the
> process of testing your Unity project. At the end of the run, the test results are parsed to JUnit
> and [stored](https://circleci.com/docs/collect-test-data) in CircleCI to enable
> [test insights](https://circleci.com/docs/insights-tests). You can access them from the CircleCI
> Dashboard.

Reviewer: ```suggestion
The Unity Orb provides the
[test job](https://circleci.com/developer/orbs/orb/game-ci/unity#jobs-test) to facilitate the
process of testing your Unity project. 

At the end of the run, the test results are parsed to JUnit
and [stored](https://circleci.com/docs/collect-test-data) in CircleCI so that [test insights](https://circleci.com/docs/insights-tests) can be used. You can access test insights from the CircleCI
Dashboard.
```

> Dashboard.

Reviewer 2: ```suggestion
web app.
```

### Parameters

> In this sub-section, you will find information on each parameter the job provides.

Reviewer: ```suggestion
Below you will find information on each parameter that can be passed to the test job.
```

#### executor

- _Default_: `N/A`
- _Required_: `True`
- _Type_: `Executor`

> 

Reviewer: Same comments from `build` apply for the descriptions here.

Author: Done ✔️

The executor on which you want to run your test job on. Each executor has its set of parameters. See
the [Executors](/docs/circleci/executors) section for more details.

```yaml
version: 2.1

orbs:
  unity: game-ci/unity@x.y

workflows:
  test-unity-project:
    jobs:
      - unity/test:
          executor:
            name: 'unity/ubuntu'
            target_platform: 'linux-il2cpp'
            editor_version: '2021.3.2f1'
            resource_class: 'medium'
```

#### project-path

- _Default_: `.`
- _Required_: `False`
- _Type_: `String`

If your project sits at the root of your repository, leave it to the default value. Otherwise, enter
the path of your Unity project. This should be the directory that has an "Assets" folder inside it.

```yaml
version: 2.1

orbs:
  unity: game-ci/unity@x.y

workflows:
  test-unity-project:
    jobs:
      - unity/test:
          project-path: 'Unity2D-Demo-Game-CI-CD/src'
```

#### step-name

- _Default_: `Run tests`
- _Required_: `False`
- _Type_: `String`

> Specify a custom step name to be shown in the CircleCI dashboard.

Reviewer 2: ```suggestion
Specify a custom step name to be shown in the CircleCI web app.
```

```yaml
version: 2.1

orbs:
  unity: game-ci/unity@x.y

workflows:
  test-unity-project:
    jobs:
      - unity/test:
          step-name: 'Test my Unity project'
```

#### test-platform

- _Default_: `editmode`
- _Required_: `False`
- _Type_: `String`

> Chose the platform to run your tests on. The available options can be found on the Unity test

Reviewer 2: ```suggestion
Choose the platform to run your tests on. The available options can be found on the Unity test
```
